dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java

```diff
@@ -0,0 +1,122 @@
+package com.dotmarketing.startup.runonce;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.dotmarketing.common.db.DotDatabaseMetaData;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+
+/**
+ * This class updates all postgres "timestamp without timezone" fields
+ * to timestamp with timezone fields.  This allows us to store dates and times 
+ * with timezone information without forcing us to conver every date/time to
+ * UTC
+ *
+ */
+public class Task210901UpdateDateTimezones extends AbstractJDBCStartupTask {
+
+
+    @Override
+    public boolean forceRun() {
+        return DbConnectionFactory.isPostgres();
+    }
+
+
+    List<String> findAllTables() throws SQLException {
+        final String[] types = {
+                "TABLE"};
+        final List<String> results = new ArrayList<>();
+        try (Connection conn = DbConnectionFactory.getConnection()) {
+            final DatabaseMetaData metaData = conn.getMetaData();
+            final ResultSet tables = metaData.getTables(null, null, "%", types);
+            while (tables.next()) {
+                results.add(tables.getString("TABLE_NAME"));
+            }
+        }
+        return results;
+
+    }
+
+
+
+    boolean updateTable(final String tableName) throws SQLException {
+
+        try (Connection conn = DbConnectionFactory.getConnection()) {
+            final ResultSet results = new DotDatabaseMetaData().getColumnsMetaData(conn, tableName);
+            while (results.next()) {
+                if ("timestamp".equals(results.getString("TYPE_NAME"))) {
```

Comment:
shouldn't be case insensitive

Reply:
shoudn't be handle as a transaction?

Reply:
I am not pretty sure about the connection handling
I know a transaction is opening on an upper layer, but the try will close this one and ask for a new one, isn't?

Reply:
> shouldn't be case insensitive

Done